File: tasks/twosum/TwoSum.java
package tasks.twosum;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numIndexes = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numIndexes.containsKey(complement)) {
                return new int[]{numIndexes.get(complement), i};
            }
            numIndexes.put(nums[i], i);
        }

        throw new IllegalArgumentException("No two sum solution");
    }
}

File: tasks/twosum/TwoSum.kt
package twosum

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val hashMap = HashMap<Int, Int>()
        for (i in nums.indices) {
            val digitToFind = target - nums[i]
            val j = hashMap[digitToFind]
            if (j != null){
                return intArrayOf(i, j)
            } 
            hashMap[nums[i]] = i
        }
            return intArrayOf()
    }
}

fun main(args: Array<String>) {
    val solution = Solution()
    for (i in solution.twoSum(intArrayOf(1,3,4,2), 6)){
    print( "$i;")
    }
}
